Add UNSIGNEDISOK variable to affect signingProxy exit code

jasonleenaylor opened the pull request:
On some CI systems we want signing to be done in a separate task so introduce a variable that we can set to return 0 from the signingProxy

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/genericinstaller/75)
<!-- Reviewable:end -->

papeh reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> when comments are addressed.

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/genericinstaller/75#-:-NsL089IBHh3frCoP1tn:b-b2y1vo)* status: all files reviewed, 3 unresolved discussions (waiting on @jasonleenaylor)

___
*[`BaseInstallerBuild/buildBaseInstaller.bat` line 2 at r1](https://reviewable.io/reviews/sillsdev/genericinstaller/75#-NsKwWey5fGCXwqcMltq:-NsKwWez0nXxVWRFBI3M:ba4atgf) ([raw file](https://github.com/sillsdev/genericinstaller/blob/dbbc767d75731041a710e5ce3e49be6bd7d98c60/BaseInstallerBuild/buildBaseInstaller.bat#L2)):*
> ```Batchfile
> @echo off
> REM Call out to the buildMsi.bat and buildExe.bat to accomplish fully signing
> ```

build and sign each part.



_Code quote:_
```Batchfile
accomplish fully signing
```
___
*[`BaseInstallerBuild/buildMsi.bat` line 45 at r1](https://reviewable.io/reviews/sillsdev/genericinstaller/75#-NsKxe7fDfk0p4ymbW5B:-NsKxe7fDfk0p4ymbW5C:b-pnw12u) ([raw file](https://github.com/sillsdev/genericinstaller/blob/dbbc767d75731041a710e5ce3e49be6bd7d98c60/BaseInstallerBuild/buildMsi.bat#L45)):*
> ```Batchfile
> 
> REM single quotes, since we expect %msbuild% is already double giquoted
> if  '%msbuild%' == '' set msbuild=msbuild
> ```

I would put this line and above in a separate SetVars.bat so that

* API changes require changes in only one script
* WiX is added to the PATH and `msbuild` is set for each script if they are run in different spaces
___
*[`BaseInstallerBuild/signingProxy.bat` line 5 at r1](https://reviewable.io/reviews/sillsdev/genericinstaller/75#-NsL-MkfA-W4CXyBp_6l:-NsL-MkfA-W4CXyBp_6m:bqhd9fc) ([raw file](https://github.com/sillsdev/genericinstaller/blob/dbbc767d75731041a710e5ce3e49be6bd7d98c60/BaseInstallerBuild/signingProxy.bat#L5)):*
> ```Batchfile
> @REM   signingProxy.bat FileToSign
> 
> if "%UNSIGNEDISOK%" == "" set UNSIGNEDISOK=1
> ```

I'd expect 0 to be false and 1 to be true. How about SIGNINGISREQUIRED or FAILIFCANTSIGN?


<!-- Sent from Reviewable.io -->

jasonleenaylor commented:
*[`BaseInstallerBuild/signingProxy.bat` line 5 at r1](https://reviewable.io/reviews/sillsdev/genericinstaller/75#-NsL-MkfA-W4CXyBp_6l:-NsLHUXl5iVwbpjpA-ih:brm01fp) ([raw file](https://github.com/sillsdev/genericinstaller/blob/dbbc767d75731041a710e5ce3e49be6bd7d98c60/BaseInstallerBuild/signingProxy.bat#L5)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

I'd expect 0 to be false and 1 to be true. How about SIGNINGISREQUIRED or FAILIFCANTSIGN?
</blockquote></details>

Changed to FAILIFCANTSIGN and added a comment explaining usage


<!-- Sent from Reviewable.io -->

jasonleenaylor reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/sillsdev/genericinstaller/75)* status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @papeh)

___
*[`BaseInstallerBuild/buildMsi.bat` line 45 at r1](https://reviewable.io/reviews/sillsdev/genericinstaller/75#-NsKxe7fDfk0p4ymbW5B:-NsPS3tX0IWizTNnF3Ta:b-896fix) ([raw file](https://github.com/sillsdev/genericinstaller/blob/dbbc767d75731041a710e5ce3e49be6bd7d98c60/BaseInstallerBuild/buildMsi.bat#L45)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

I would put this line and above in a separate SetVars.bat so that

* API changes require changes in only one script
* WiX is added to the PATH and `msbuild` is set for each script if they are run in different spaces
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->
